biojava-core/src/main/java/org/biojava/nbio/core/sequence/io/embl/EmblReader.java

```diff
@@ -0,0 +1,147 @@
+package org.biojava.nbio.core.sequence.io.embl;
+
+
+import java.io.*;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * This class should process the data of embl file
+ * @since 5.0.0
+ * @author Noor Aldeen Al Mbaidin
+ */
+public class EmblReader {
+
+    private StringBuilder sequence = new StringBuilder("");
+
+    public EmblReader() {
+
+    }
+
+    /**
+     * The parsing is done in this method.<br>
+     * This method tries to process all the Embl records
+     * in the File , closes the underlying resource,
+     * and return the results in object of EmblRecord.<br>
+     *
+     * @return EmblRecord containing all the parsed Embl records
+     * @throws IOException
+     */
+    public static EmblRecord process(File file) throws IOException {
+
+        EmblRecord emblRecord = new EmblRecord();
+        StringBuilder sequence = new StringBuilder("");
+        EmblReference emblReference = new EmblReference();
+        LinkedList<String> accessionNumber = new LinkedList<>();
+        LinkedList<String> keyword = new LinkedList<>();
+
+        if (file == null)
+            throw new NullPointerException("file can't be null");
+
+        if(file.isDirectory())
+            throw new IllegalArgumentException("the file can't be a directory");
+
+        try (FileReader fileReader = new FileReader(file)) {
+            String line = "";
+            String lineIdentifier;
+            String lineInfo;
+            try (BufferedReader bufferedReader = new BufferedReader(fileReader)) {
+                while ((line = bufferedReader.readLine()) != null) {
+                    lineInfo = line.substring(0, 2);
```

Comment:
What is the difference between `lineInfo` and `lineIdentifier` here?  Note also that there is no check if `line` is long enough for this substring call not to fail.

Reply:
Sorry this was a mistake , I have added validation for line length.